api/lib/opentelemetry/instrumentation/base.rb

```diff
@@ -180,12 +186,31 @@ def install(config = {})
         return true if installed?
         return false unless installable?(config)
 
-        @config = config unless config.nil?
+        @config = config_options(config)
         instance_exec(@config, &@install_blk)
         @tracer ||= OpenTelemetry.tracer_provider.tracer(name, version)
         @installed = true
       end
 
+      def config_options(config_hash)
+        config_hash ||= {}
+
+        @options.each_with_object({}) do |option, h|
+          option_name = option[:name]
+          config_hash_value = config_hash[option_name]
+
+          value = if config_hash_value.nil? && option[:allow_nil]
+            nil
+          elsif option[:validate].call(config_hash_value)
```

Comment:
In the same way we're providing a validation of config options here, we  shouuld probably validate that this is callable, or a proc, or something to that effect? Wdyt?

Reply:
I think that's a reasonable thing to do, tests would only catch that error if there is test explicitly setting that config option.